src/game/SpellEffects.cpp

```diff
@@ -2572,14 +2556,20 @@ void Spell::EffectSummonGuardian(SpellEffectIndex eff_idx)
         if (m_caster->FindGuardianWithEntry(pet_entry))
             return;                                         // find old guardian, ignore summon
 
-    // in another case summon new
-    uint32 level = m_caster->getLevel();
+    // Level of pet summoned
+    uint32 level = std::max(m_caster->getLevel() + m_spellInfo->EffectMultipleValue[eff_idx], 1.0f);
```

Comment:
1.0f (float) should be just 1 (integer) actually, otherwise std::max for float values is called and the return type of std::max would be float too (which leads to unneeded implicit conversions from float to int). Just a small detail.